src/wp-includes/class-wp-theme.php

```diff
@@ -1840,6 +1798,213 @@ public function get_block_template_folders() {
 		return $this->block_template_folders;
 	}
 
+	/**
```

Comment:
Reviewer note:

These methods are just moved from above, so they are organized with other block-related methods. The only changes are that the `set_pattern_cache` and `get_pattern_cache` are now private, while the `delete_pattern_cache` remains public for cache clearing purposes.